### PR TITLE
More integration test cluster logging improvements

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

ADP-302 / #1750

### Overview

- Output Debug-severity log files for each cardano-node process to the temporary directory, while only showing Error-severity logs on stdout.
- Show different hostnames in cardano-node logs so that it's possible to see which node produced the log message.
- Add a new helper function for starting the cardano-node process.
- Use a single temp directory for all nodes in the cluster - makes inspecting configs and logs easier.
- The environment variable `NO_CLEANUP` causes the integration tests to not delete the temp directory after they have finished.
- The environment variable `CARDANO_NODE_TRACING_MIN_SEVERITY` sets the log level the the cluster nodes for stdout.
- The environment variable `CARDANO_WALLET_TRACING_MIN_SEVERITY` sets the log level for the wallet server.
- Also bumps cardano-node and libraries to latest master (to get latest logging features).

### CODE REVIEW

**Comment by rvl** [comment]
This is rebased and ready for review.

File: lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs [hunk 1]
```diff
@@ -334,56 +334,75 @@ data NodeConfig = NodeConfig
...
+                , nodeTopologyFile = topology
+                , nodeDatabaseDir = "db"
+                , nodeDlgCertFile = Nothing
+                , nodeSignKeyFile = Nothing
+                , nodeOpCertFile = Just opCert
+                , nodeKesKeyFile = Just kesPrv
+                , nodeVrfKeyFile = Just vrfPrv
+                , nodePort = Just (NodePort port)
+                , nodeLoggingHostname = Just name
+                }
```
Comment (KtorZ) [inline lib/shelley/src/Cardano/Wallet/Shelley/Launch.hs:369]: :+1:

**Review by KtorZ — APPROVED** [review]
LGTM

**Pull request merged**